Add ufs-utils template

HenryRWinterbottom opened the pull request:
### Summary

This PR, in conjunction with spack PR [#300](https://github.com/JCSDA/spack/pull/300) provides stack building support required to build [UFS_UTILS](https://github.com/ufs-community/UFS_UTILS).

### Testing

This PR has been compiled on both the NOAA CSP AWS PW platform and RDHPCS Hera. The resulting stack supports the compilation of the UFS_UTILS applications, including the available unit-tests.

### Applications affected

No applications are impacted. UFS_UTILS is now supported. 

### Systems affected

None.

### Dependencies

- https://github.com/JCSDA/spack/pull/300

### Issue(s) addressed

This PR addresses issue #644.

### Checklist
- [x] This PR addresses one issue/problem/enhancement, or has a very good reason for not doing so.
- [x] These changes have been tested on the affected systems and applications.
- [x] All dependency PRs/issues have been resolved and this PR can be merged.

HenryRWinterbottom committed:
Spack configuration for UFS_UTILS.

HenryRWinterbottom committed:
Merge branch 'develop' into feature/ufs_utils_template

HenryRWinterbottom committed:
Merge branch 'develop' into feature/ufs_utils_template

HenryRWinterbottom committed:
Updating to comply with JCSDA implementation.

HenryRWinterbottom committed:
Merge branch 'JCSDA:develop' into subfeature/packages_py

HenryRWinterbottom committed:
Merge branch 'JCSDA:develop' into subfeature/packages_py

Commit:
Updated missing version specific packages.

climbfuji commented:
@HenryWinterbottom-NOAA I merged the spack PR. The new hash for the spack submodule is https://github.com/JCSDA/spack/commit/24657b60ad3ea50c99a1d68de8c178a410ba54eb. Please:
- update the spack submodule pointer in spack-stack to point to this hash
- pull in the updates from spack-stack develop

Thanks!

HenryRWinterbottom commented:
@climbfuji Would you like me to create a new branch/issue for this?

climbfuji commented:
No, you just need to push these updates to this branch

HenryRWinterbottom committed:
Merge branch 'JCSDA:develop' into subfeature/packages_py

HenryRWinterbottom committed:
Submodule update.

HenryRWinterbottom commented:
> No, you just need to push these updates to this branch

@climbfuji I made the requested updates. Please let me if I misunderstood and/or missed something.

climbfuji commented on .gitmodules:
```diff
@@ -3,7 +3,7 @@
   #branch = develop
   url = https://github.com/jcsda/spack
-  branch = jcsda_emc_spack_stack
+  branch = JCSDA/spack@24657b6
```
This change needs to be reverted. You need to update the submodule pointer, not `.gitmodules`.

HenryRWinterbottom replied:
@climbfuji Can you point me to the correct file and/or give me explicit instructions? I don't know what the "submodule pointer" is referring to.

climbfuji replied:
In your local spack-stack clone of your branch `subfeature/packages_py`:
```
cd spack
git remote update
git checkout origin/jcsda_emc_spack_stack
git log
# this should show you the hash I mentioned at the very top
cd ..
git diff
# you should see a diff showing up for "spack (new commits)"
git add spack
git commit -m "Update submodule pointer for spack"
git push YOUR_REMOTE_NAME subfeature/packages_py
```

HenryRWinterbottom replied:
@climbfuji OK. I just submitted attempt number 2.

climbfuji replied:
submodule pointer is looking good now, just need to revert `.gitmodules`.

HenryRWinterbottom replied:
To save headaches, how should I properly do that given our current state of the repo?

HenryRWinterbottom replied:
@climbfuji OK, reverted. 

Thank you for your patience with me.

climbfuji replied:
CI tests are running now, thanks!

climbfuji replied:
They shouldn't be affected by your changes, but I'll let them finish nonetheless before merging.

climbfuji reviewed (changes requested):
Let me know if you need my help with this.

HenryRWinterbottom committed:
Update submodule pointer for spack.

climbfuji commented on .gitmodules:
```diff
@@ -3,7 +3,7 @@
   #branch = develop
   url = https://github.com/jcsda/spack
-  branch = jcsda_emc_spack_stack
+  branch = JCSDA/spack@24657b6
```
```suggestion
  branch = jcsda_emc_spack_stack
```

HenryRWinterbottom replied:
@climbfuji No problem at all! I am happy to contribute and thank you again  for the opportunity.

HenryRWinterbottom committed:
Reverted change.

climbfuji commented on configs/templates/ufs-utils/spack.yaml:
```diff
@@ -0,0 +1,43 @@
+  specs:
+    - ufs-utils-env~debug  
+
+  view: false
```
The CI unit tests found an error:
```
FAILED tests/test_stack_create.py::test_apps[ufs-utils] - spack.util.spack_yaml.SpackYAMLError: error parsing YAML: near /home/runner/work/spack-stack/spack-stack/configs/templates/ufs-utils/spack.yaml, 1, 2: found duplicate key "view" with value "False" (original value: "False")
```

Please remove lines 20 and 21 (just for the sake of not having unnecessary blank lines), hopefully that will do.

HenryRWinterbottom replied:
@climbfuji I just make the correction. Please see my note in the commit.

HenryRWinterbottom committed:
Update spack.yaml

I removed the duplicate `specs` and `view` keys. 

If this doesn't conform to the YAML format standards, i.e., the spacings beneath the `specs` key list, I can fix that in a future revision.

climbfuji commented on configs/templates/ufs-utils/spack.yaml:
```diff
@@ -0,0 +1,39 @@
+  - jasper@2.0.25
+  - bufr@11.7.0
+  - wgrib2@2.0.8
+  include:
```
You should remove those includes, they will get added automatically by `spack stack create env`. Basically, your file should look like https://github.com/JCSDA/spack-stack/blob/develop/configs/templates/empty/spack.yaml with the specs section being expanded.

HenryRWinterbottom replied:
@climbfuji Done.

HenryRWinterbottom committed:
Update spack.yaml
